### PR TITLE
Fix travis test

### PR DESCRIPTION
**Pull request by yuwzho**
Refer this [issue](https://github.com/Azure-Samples/iot-hub-c-thingdev-getstartedkit/issues/25). Some linux system cannot recognize this backslah
@mamokarz

### CODE REVIEW

**Comment by yuwzho** [comment]
I also make some changes for the .travis.yml, which install some necessary libraries and let the CI for ESP8266 work. But the samd boards still have some header not found

**Comment by yuwzho** [comment]
Fix the travis, the samd boards should use at least Arduino 1.6.11 IDE

**Comment by tameraw** [comment]
@yuwzho - We have fixed the slash for Linux in the version 1.0.30 as part of a separate check-in we had to do. We will review the other changes and get back to you.

**Comment by yuwzho** [comment]
@tameraw Thank you! 

These pull request also includes install nessary libraries for Travis CI testing. Since some board definition updating, the new board package and old Arduino IDE are not compatiable. So I update the IDE version. 

Also, the `Serial.setDebugOutput(boolean);` is a function for ESP8266, not for all boards.

**Comment by yuwzho** [comment]
Rebase the pullrequest and update the title

**Comment by JetstreamRoySprowl** [comment]
Hi Yuwei, we're about to do the merge and were wondering 

1. What is the motivation for only putting the Serial.setDebugOutput(true) call in for ESP8266? A comment might be good here.
2. Line 24 of the .yml shows a redundant call to install the WiFi101 library. If there is a reason this needs to happen twice then a comment would definitely be useful.

File: .travis.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ before_install:
   - if [[ "$BOARD" =~ "arduino:samd:" ]]; then
       arduino --install-boards arduino:samd;
       arduino --install-library WiFi101;
```
Comment (tameraw) [inline .travis.yml]: >arduino --install-library WiFi101; [](start = 6, length = 34)

Appears duplicate of library below.

Comment (yuwzho) [inline .travis.yml]: An interesting thing is the first library install always failed with message "selected libraries not found", no idea why this issue happen on travis

Comment (tameraw) [inline .travis.yml]: Got it. We plan to merge in this PR, but we will remove the #ifdef around the Serial.setDebugOutput(true). If this is blocking you in any way, please let us know.

Comment (yuwzho) [inline .travis.yml]: @tameraw The `Serial.setDebugOutput(true)` is only for 8266 board. It is an interface defined in the 8266 board package, not for all boards.

You can ref these pages:
http://www.panstamp.org/forum/archive/index.php/thread-4541.html 
http://arduino.esp8266.com/versions/1.6.5-1044-g170995a/doc/reference.html#serial

**Comment by JetstreamRoySprowl** [comment]
We've accepted the pull request and it will be merged into the next release.

**Pull request closed without merging**